src/extended/api.py

```diff
@@ -0,0 +1,73 @@
+# todo change to module imports
+import time
+from dataclasses import asdict, dataclass, field
+from typing import Any, List, Union
+
+from extended.models import Case, Section, Suite
+from testrail import APIClient
+
+
+class TestRail(APIClient):
+    # it would make a lot of sense to move these into their respective models and then have mixins to create apis,
+    # this would allow you to see what the dependneices are between you and the other API calls you want o amke
+    # todo, ensure all typing is done - this will only be python3 compatable.
+    def __init__(self, user, password, url):
+        super().__init__(user, password, url)
+
+    def get_case(self, case_id):
+        return self.send_get(f"get_case/{case_id}")
+
+    def get_case_types(self):
+        return self.send_get("get_case_types")
+
+    def get_priorities(self):
+        return self.send_get("get_priorities")
+
+    def get_case_fields(self):
+        return self.send_get("get_case_fields")
+
+    def get_suite(self, suite_id):
+        suite = Suite(**self.send_get(f"get_suite/{suite_id}"))
+        sections = self.get_sections(suite.project_id, suite.id)
+        return (
+            suite,
+            sections[0],
+        )  # hopefully we can just have one setion for each suite for now, what is gonna be the best way to
+        # eventually it would be nice to be able to setup
+
+    def get_suites(self, project_id):
+        suites = [Suite(**_) for _ in self.send_get(f"get_suites/{project_id}")]
+        return suites
+
+    def add_suite(self, name: str, description: str, project_id: int):
```

Comment:
definitely need to be able to provide a list of sections to this as a param, if None, create the default.
might make sense as a better helper function to keep the api here pure